scripts/app.js

```diff
@@ -14,18 +16,49 @@ Project.prototype.toHTML = function(template) {
   return template(this);
 };
 
-portfolioProjects.sort(function(a,b) {
-  return (new Date(b.postDate)) - (new Date(a.postDate));
-});
-
-portfolioProjects.forEach(function(ele) {
-  projects.push(new Project(ele));
-});
+Project.loadAll = function(data) {
+  data.sort(function(a,b) {
+    return (new Date(b.postDate)) - (new Date(a.postDate));
+  });
+  data.forEach(function(ele) {
+    Project.all.push(new Project(ele));
+  });
+};
 
-projects.forEach(function(a) {
-  $('#projects').append(a.toHTML('#project-template'));
-  if (categories.indexOf(a.category)) {
-    $('.filters ul').append(a.toHTML('#category-filter-template'));
-    categories.push(a.category);
-  };
-});
+Project.fetchAll = function() {
+  localStorage.eTag = JSON.stringify(0);
```

Comment:
This line here will always reset your localStorage.etag.  So when you check to see if it matches the non-local etag it will never match and you will always reload the data.